### PR TITLE
Add auth to more server routes

### PR DESCRIPTION
**Pull request by jacobkrantz**
@mdulin2, what should be done for `/api/v1/users/add` and `/api/v1/users/login` routes? Left as-is?

### CODE REVIEW

**Comment by mdulin2** [comment]
@jacobkrantz Insert should be an admin only route (creating a user). Login should **not** be authenticated.

**Comment by jacobkrantz** [comment]
@mdulin2 please check this :smile:

File: WebPhapp/backend/server.js [hunk 1]
```diff
@@ -258,7 +258,7 @@ Returns:
         prescriberID, dispenserID, cancelDate, drugName
     ]
 */
-app.get('/api/v1/prescriptions/:patientID', (req,res) => {
+app.get('/api/v1/prescriptions/:patientID', auth.checkAuth([Role.Patient, Role.Prescriber, Role.Government]), (req,res) => {
```
Comment (mdulin2) [inline WebPhapp/backend/server.js]: Should dispensers be able to see this information? I think they should be able to.

File: WebPhapp/backend/server.js [hunk 2]
```diff
@@ -362,7 +362,7 @@ Returns:
 Warning:
     no validation exists for blockchain index yet. See Issue #32 on GitHub.
 */
-app.get('/api/v1/prescriptions/single/:prescriptionID', (req,res) => {
+app.get('/api/v1/prescriptions/single/:prescriptionID', auth.checkAuth([Role.Patient, Role.Prescriber, Role.Government, Role.Dispenser]), (req,res) => {
```
Comment (mdulin2) [inline WebPhapp/backend/server.js]: Will need validation in the patient role.

File: WebPhapp/backend/server.js [hunk 3]
```diff
@@ -460,7 +460,7 @@ Returns:
 Relevant Express Docs:
     https://expressjs.com/en/api.html#req.query
 */
-app.get('/api/v1/patients', (req,res) => {
+app.get('/api/v1/patients', auth.checkAuth([Role.Prescriber, Role.Government, Role.Dispenser]), (req,res) => {
```
Comment (mdulin2) [inline WebPhapp/backend/server.js]: Patients should be able to see information about themselves? Or, are we not allowing this ?

File: WebPhapp/backend/server.js [hunk 4]
```diff
@@ -555,8 +555,10 @@ Examples:
     curl 'http://localhost:5000/api/v1/users/add' -H 'Acceptapplication/json, text/plain, /*' -H 'Content-Type: application/json;charset=utf-8' --data '{"username":"mdulin2","password":"jacob","role":"Patient"}'
 Returns:
     Success or failure with message
+Authentication:
+    Admin only (Role.Admin)
 */
-app.post('/api/v1/users/add', (req,res) => {
+app.post('/api/v1/users/add', auth.checkAuth([]), (req,res) => {
```
Comment (mdulin2) [inline WebPhapp/backend/server.js]: This looks funny; but, is the correct way to go about this.

File: WebPhapp/backend/server.js (hunk 2 above)
Comment (jacobkrantz) [reply WebPhapp/backend/server.js]: I don't know how to test your JWT stuff yet, so I don't know how to extract PatientID from the request (if it is even present yet).

**Pull request merged**